pkg/controller/perconaservermongodb/mgo.go

```diff
@@ -59,6 +59,24 @@ func (r *ReconcilePerconaServerMongoDB) reconcileCluster(cr *api.PerconaServerMo
 		return clusterError, errors.Wrap(err, "dial:")
 	}
 
+	if cr.Status.Replsets[replset.Name].Initialized &&
+		cr.Status.Replsets[replset.Name].Status == api.AppStateReady &&
+		replset.ClusterRole == api.ClusterRoleShardSvr &&
```

Comment:
Do we need these cluster roles at all?

Reply:
Yes, because operator does reconcile over all replicas. And one of this replica is configServer.